src/Sign.Core/SignatureProviders/ClickOnceSignatureProvider.cs

```diff
@@ -129,19 +129,26 @@ public async Task SignAsync(IEnumerable<FileInfo> files, SignOptions options)
                             throw new Exception(message);
                         }
 
-                        // Read the publisher name from the manifest for use below
-                        XDocument manifestDoc = XDocument.Load(manifestFile.FullName);
-                        XNamespace ns = manifestDoc.Root!.GetDefaultNamespace();
-                        XElement? publisherIdentity = manifestDoc.Root.Element(ns + "publisherIdentity");
-                        string publisherName = publisherIdentity!.Attribute("name")!.Value;
+                        string publisherParam = "";
 
-                        var publisherParam = "";
-
-                        Dictionary<string, List<string>> dict = DistinguishedNameParser.Parse(publisherName);
-                        if (dict.TryGetValue("CN", out var cns))
+                        if (string.IsNullOrEmpty(options.PublisherName))
+                        {
+                            // Read the publisher name from the manifest for use below
+                            XDocument manifestDoc = XDocument.Load(manifestFile.FullName);
+                            XNamespace ns = manifestDoc.Root!.GetDefaultNamespace();
+                            XElement? publisherIdentity = manifestDoc.Root.Element(ns + "publisherIdentity");
+                            string publisherName = publisherIdentity!.Attribute("name")!.Value;
+                            Dictionary<string, List<string>> dict = DistinguishedNameParser.Parse(publisherName);
+ 
+                            if (dict.TryGetValue("CN", out var cns))
```

Comment:
Same here for `"CN"`, probably better to create const for it.

Reply:
See my previous comment.